### PR TITLE
Issue #674 Slightly re-org Dockefile to improve layer caching

### PR DESCRIPTION
**Pull request by jonespm**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -25,6 +22,11 @@ RUN npm prune --production && \
 # FROM directive instructing base image to build upon
 FROM python:3.6 AS app
 
+# Sets the local timezone of the docker image
+ARG TZ
+ENV TZ ${TZ:-America/Detroit}
+RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
+
```
Comment (pushyamig) [inline Dockerfile]: what is the benefit of moving this `TZ` setting here?

File: Dockerfile [hunk 2]
```diff
@@ -36,21 +38,17 @@ RUN apt-get update && \
...
-COPY . /code/
 # copy built react and node libraries for production mode
 COPY --from=node-prod-deps /usr/src/app/package-lock.json /code/package-lock.json
 COPY --from=node-prod-deps /usr/src/app/webpack-stats.json /code/webpack-stats.json
 COPY --from=node-prod-deps /usr/src/app/assets /code/assets
 COPY --from=node-prod-deps /usr/src/app/node_modules /code/node_modules
 
+# NOTE: project files likely to change between dev builds
+COPY . /code/
+
```
Comment (pushyamig) [inline Dockerfile]: Why this line needed to be moved?

**Comment by pushyamig** [comment]
This looks Good.

File: Dockerfile (hunk 1 above)
Comment (jonespm) [reply Dockerfile]: I think moving these up higher is actually wrong, I'll move it back down where it was.
https://medium.com/@esotericmeans/optimizing-your-dockerfile-dc4b7b527756

File: Dockerfile (hunk 2 above)
Comment (jonespm) [reply Dockerfile]: The main part of this was moving the COPY of code down below the copy of the node dependencies. That takes a little time to copy that and those don't change as much as the code. You can look at this as move the 4 COPY lines for --from=node-prod-deps above this one. ;)

**Pull request merged**